File: go/fibonacci.go
package main

import (
    "fmt"
    "time"
)

func fibonacci(n int) int {
    if n <= 1 {
        return n
    }
    return fibonacci(n-1) + fibonacci(n-2)
}

func fibonacciIterative(n int) int {
    if n <= 1 {
        return n
    }
    
    a, b := 0, 1
    for i := 2; i <= n; i++ {
        a, b = b, a+b
    }
    return b
}

func main() {
    n := 40
    
    // Recursive version
    start := time.Now()
    result := fibonacci(n)
    recursiveDuration := time.Since(start)
    
    fmt.Printf("Go Recursive Fibonacci(%d) = %d\n", n, result)
    fmt.Printf("Time: %v\n", recursiveDuration)
    
    // Iterative version
    start = time.Now()
    result = fibonacciIterative(n)
    iterativeDuration := time.Since(start)
    
    fmt.Printf("Go Iterative Fibonacci(%d) = %d\n", n, result)
    fmt.Printf("Time: %v\n", iterativeDuration)
}

File: rust/src/fibonacci.rs
use std::time::Instant;

fn fibonacci(n: u32) -> u32 {
    if n <= 1 {
        return n;
    }
    fibonacci(n - 1) + fibonacci(n - 2)
}

fn fibonacci_iterative(n: u32) -> u32 {
    if n <= 1 {
        return n;
    }
    
    let mut a = 0;
    let mut b = 1;
    
    for _ in 2..=n {
        let temp = a + b;
        a = b;
        b = temp;
    }
    
    b
}

fn main() {
    let n = 40;
    
    // Recursive version
    let start = Instant::now();
    let result = fibonacci(n);
    let recursive_duration = start.elapsed();
    
    println!("Rust Recursive Fibonacci({}) = {}", n, result);
    println!("Time: {:?}", recursive_duration);
    
    // Iterative version
    let start = Instant::now();
    let result = fibonacci_iterative(n);
    let iterative_duration = start.elapsed();
    
    println!("Rust Iterative Fibonacci({}) = {}", n, result);
    println!("Time: {:?}", iterative_duration);
}